### PR TITLE
fix: send search query event (intelligent search)

### PR DESCRIPTION
**Pull request by icazevedo**
## What's the purpose of this pull request?

This PR fixes IS reports for search queries

## How does it work?

Adds `search.query` event to IS integration, fired whenever a search query with `term` is executed.

## How to test it?

FIlter network requests by `sp` and look for the requests to IS with the new `search.query` event.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,32 +30,32 @@
     "@envelop/graphql-jit": "^1.1.1",
     "@envelop/parser-cache": "^2.2.0",
     "@envelop/validation-cache": "^2.2.0",
-    "@faststore/api": "^1.12.29",
+    "@faststore/api": "https://pkg.csb.dev/vtex/faststore/commit/3904aba5/@faststore/api",
```
Comment (emersonlaurentino) [inline package.json:33]: 👀

**Review by emersonlaurentino — APPROVED** [review]

File: src/sdk/cart/index.ts [hunk 2]
```diff
@@ -119,6 +120,7 @@ const validateCart = async (cart: Cart): Promise<Cart | null> => {
             },
           })
         ),
+        shouldSplitItem: false,
```
Comment (eduardoformiga) [inline src/sdk/cart/index.ts]: is it intentional?

**Review by eduardoformiga — APPROVED** [review]
Good job!
I manage to simulate the search event 
Screen Shot 2023-02-13 at 17 32 15

File: src/sdk/cart/index.ts (hunk 2 above)
Comment (icazevedo) [reply src/sdk/cart/index.ts]: It was! In version 2.x, this change was introduced. But as we reverted back to using 1.x, it was no longer needed, so I removed it.

**Pull request merged**